src/qibocal/protocols/resonator_utils.py

```diff
@@ -0,0 +1,360 @@
+        )
+        * m_matrix[3][0]
+        + (
+            (m_matrix[2][3] * m_matrix[3][1] - m_matrix[2][1] * m_matrix[3][3])
+            * m_matrix[1][2]
+            + m_matrix[2][1] * m_matrix[3][2] * m_matrix[1][3]
+        )
+        * m_matrix[0][0]
+        + (
+            m_matrix[1][0] * m_matrix[2][3] * m_matrix[3][2]
+            + m_matrix[2][0]
+            * (m_matrix[1][2] * m_matrix[3][3] - m_matrix[1][3] * m_matrix[3][2])
+        )
+        * m_matrix[0][1]
+        + (
+            (m_matrix[2][1] * m_matrix[3][3] - m_matrix[2][3] * m_matrix[3][1])
+            * m_matrix[1][0]
+            + m_matrix[1][3] * m_matrix[2][0] * m_matrix[3][1]
+        )
+        * m_matrix[0][2]
+    )
+    a_1 = (
+        (
+            (m_matrix[3][0] - 2.0 * m_matrix[2][2]) * m_matrix[1][1]
+            - m_matrix[1][0] * m_matrix[3][1]
+            + m_matrix[2][2] * m_matrix[3][0]
+            + 2.0 * m_matrix[1][2] * m_matrix[2][1]
+            - m_matrix[2][0] * m_matrix[3][2]
+        )
+        * m_matrix[0][3]
+        + (
+            2.0 * m_matrix[2][0] * m_matrix[3][2]
+            - m_matrix[0][0] * m_matrix[3][3]
+            - 2.0 * m_matrix[2][2] * m_matrix[3][0]
+            + 2.0 * m_matrix[0][2] * m_matrix[2][3]
+        )
+        * m_matrix[1][1]
+        + (
+            -m_matrix[0][0] * m_matrix[3][3]
+            + 2.0 * m_matrix[0][1] * m_matrix[1][3]
+            + 2.0 * m_matrix[1][0] * m_matrix[3][1]
+        )
+        * m_matrix[2][2]
+        + (
+            -m_matrix[0][1] * m_matrix[1][3]
+            + 2.0 * m_matrix[1][2] * m_matrix[2][1]
+            - m_matrix[0][2] * m_matrix[2][3]
+        )
+        * m_matrix[3][0]
+        + (m_matrix[1][3] * m_matrix[3][1] + m_matrix[2][3] * m_matrix[3][2])
+        * m_matrix[0][0]
+        + (m_matrix[1][0] * m_matrix[3][3] - 2.0 * m_matrix[1][2] * m_matrix[2][3])
+        * m_matrix[0][1]
+        + (m_matrix[2][0] * m_matrix[3][3] - 2.0 * m_matrix[1][3] * m_matrix[2][1])
+        * m_matrix[0][2]
+        - 2.0 * m_matrix[1][2] * m_matrix[2][0] * m_matrix[3][1]
+        - 2.0 * m_matrix[1][0] * m_matrix[2][1] * m_matrix[3][2]
+    )
+    a_2 = (
+        (2.0 * m_matrix[1][1] - m_matrix[3][0] + 2.0 * m_matrix[2][2]) * m_matrix[0][3]
+        + (2.0 * m_matrix[3][0] - 4.0 * m_matrix[2][2]) * m_matrix[1][1]
+        - 2.0 * m_matrix[2][0] * m_matrix[3][2]
+        + 2.0 * m_matrix[2][2] * m_matrix[3][0]
+        + m_matrix[0][0] * m_matrix[3][3]
+        + 4.0 * m_matrix[1][2] * m_matrix[2][1]
+        - 2.0 * m_matrix[0][1] * m_matrix[1][3]
+        - 2.0 * m_matrix[1][0] * m_matrix[3][1]
+        - 2.0 * m_matrix[0][2] * m_matrix[2][3]
+    )
+    a_3 = (
+        -2.0 * m_matrix[3][0]
+        + 4.0 * m_matrix[1][1]
+        + 4.0 * m_matrix[2][2]
+        - 2.0 * m_matrix[0][3]
+    )
+    a_4 = -4
+
+    def pol_func(x: float) -> float:
+        """
+        Polynomio for find a root of a real or complex function using the Newton-Raphson method.
```

Comment:
```suggestion
        Polynomial for find a root of a real or complex function using the Newton-Raphson method.
```

The word *polynomio* is certainly not English...